Add a custom GL/Rhi adapter for systems without OpenGL

Kidev opened the pull request:
- Add a `RHI` option to the `CMakeLists.txt` that controls if the graphic API is OpenGL or Rhi
- Create `OpenGLRHIAdapter.h` that is able to give the correct graphic API objects to the requesting widgets in `src/widgets` depending on the value of the `RHI` option
- The widgets are untouched, they just required the change of any of their include of a file in `src/3d/` to `src/3d/OpenGLRHIAdapter.h`
- The `build.yml` is adapted so that it always build with `RHI=OFF`, unless the arch is Windows ARM64
- Fix all warnings
- Add Rhi (SPIR-V) shaders pre-built

Kidev commented:
Hmm that was my main fear.

> Any idea what I can debug here to help you out?

Well, it's simple really: how can I test on my own? I don't have the game. I tried to download some mods in the hope that I could have something to load in any of the tab that require a custom render

Kidev commented:
Did not see your edit, I'll try

Kidev commented:
Alright so @julianxhokaxhiu to update you on QRhi I have a display, with working camera (sliders+arrows), background, reset, inside the Walkmesh. Just need to display what's supposed to be displayed and not a static triangle and I'll update my branch

julianxhokaxhiu commented:
Let me give it a try and I'll post some screenshots to compare, but great job so far :) Thanks a lot!

Kidev commented:
@julianxhokaxhiu I did not update my branch here yet, but I'm doing so pretty soon

Gonna take a look at the worldmap! For the walkmesh, I fixed the cyan borders, fixed the z fighting, fixed the overwriting of orange edges by accumulating edges before rendering them (so that the rest of the mesh does not rewrite a orange edge into a white one, this is a bug fix compared to the OpenGL version), the squares are only displayed when the corresponding tab is selected (Exits/Doors), but the squares around the selected triangle are always shown
<img width="2558" height="1440" alt="2025-08-09_23-36" src="https://github.com/user-attachments/assets/565440d6-9b4d-4c4a-897c-e66b83db7bd1" />

Kidev commented:
Last thing I'm "worried" about is that the coordinate system setup in the new API might cause hidden bugs (X,Y are flipped)

Kidev commented:
@julianxhokaxhiu [New PR here](https://github.com/Kidev/deling/pull/5)
